### ISSUE TITLE
aioboto3's `BatchWriter` inferred as `object`

### ISSUE THREAD

**Issue by a14n** [astral-sh/ty#3892]
### Summary

The following code produces a diagnostic because `table.batch_writer()` is seen as `object` rather than `BatchWriter`
```py
from types_aiobotocore_dynamodb.service_resource import Table


async def truncate_table(table: Table) -> None:
    async with table.batch_writer() as batch:
        pass
```

Here is the diagnostic:

```
error[invalid-context-manager]: Object of type `object` cannot be used with `async with` because it does not implement `__aenter__` and `__aexit__`
 --> src/my_package/issue.py:5:16
  |
5 |     async with table.batch_writer() as batch:
  |                ^^^^^^^^^^^^^^^^^^^^
  |

Found 1 diagnostic
```

The stub source is:

```py
    def batch_writer(
        self,
        overwrite_by_pkeys: list[str] | None = ...,
        flush_amount: int = ...,
        on_exit_loop_sleep: int = ...,
    ) -> BatchWriter:
```

And the dependencies:

```
dependencies = [
  "aioboto3",
  "types-aioboto3[dynamodb,s3,sqs,stepfunctions]",
]
```

### Version

ty 0.0.55 (42869400d 2026-06-26)

**Comment by sharkdp** [comment]
Thank you for reporting this.

ty does find the stub file, but the `service_resource.pyi` stub has these problematic lines:

```py
try:
    from aioboto3.dynamodb.table import BatchWriter
except ImportError:
    from builtins import object as BatchWriter  # type: ignore[assignment]
```

This is why ty sees `BatchWriter` as `object`. I think this should be fixed in `types-aioboto3`.

**Comment by carljm** [comment]
I think this is the same as #2096, which describes a possible workaround we could use in ty to handle this pattern better. But given that we only ever seem to see this reported from boto3 stubs, it seems like a a fix there might make more sense.
